config/src/main/scala/com/lynbrookrobotics/potassium/config/TwoWaySignal.scala

```diff
@@ -1,33 +1,65 @@
 package com.lynbrookrobotics.potassium.config
 
+import scala.ref.WeakReference
+
+case class ChildReference[T, U](child: WeakReference[TwoWaySignal[U]], transform: T => U) {
+  type TargetType = U
+
+  val tt: WeakReference[TwoWaySignal[TargetType]] = child.asInstanceOf[WeakReference[TwoWaySignal[TargetType]]]
+}
+
 trait TwoWaySignal[T] { self =>
-  def value: T
+  private var currentValue: T = null.asInstanceOf[T]
+
+  private var children: List[ChildReference[T, _]] = List.empty
 
-  def reversePropagate(newValue: T): Unit
+  def value: T = currentValue
+
+  def handlePropogate(newValue: T): Unit
+
+  def reversePropagate(newValue: T): Unit = {
+    currentValue = newValue
+    handlePropogate(newValue)
+  }
 
   def map[U](transform: T => U)(reverseTransform: (T, U) => T): TwoWaySignal[U] = {
-    new TwoWaySignal[U] {
-      override def reversePropagate(newValue: U): Unit = {
+    val mappedSignal = new TwoWaySignal[U] {
+      override def handlePropogate(newValue: U): Unit = {
         self.reversePropagate(reverseTransform(self.value, newValue))
       }
+    }
+
+    children = ChildReference(new WeakReference(mappedSignal), transform) :: children
 
-      override def value: U = transform(self.value)
+    mappedSignal.updateNoReverse(transform(currentValue))
+
+    mappedSignal
+  }
+
+  private[TwoWaySignal] def updateNoReverse(newValue: T): Unit = {
+    if (currentValue != newValue) {
+      currentValue = newValue
+
+      children = children.flatMap { cr =>
+        cr.tt.get.map { child =>
+          child.updateNoReverse(cr.transform(currentValue).asInstanceOf[cr.TargetType])
+          cr
+        }
+      }
     }
   }
 
   def value_=(newValue: T): Unit = {
-    reversePropagate(newValue)
+    if (currentValue != newValue) {
+      updateNoReverse(newValue)
+      reversePropagate(newValue)
+    }
   }
 }
 
 object TwoWaySignal{
   def apply[T](initialValue: T): TwoWaySignal[T] = new TwoWaySignal[T] {
-    private var currentValue = initialValue
-
-    override def value: T = currentValue
-
-    override def reversePropagate(newValue: T): Unit = {
-      currentValue = newValue
+    override def handlePropogate(newValue: T): Unit = {
```

Comment:
Just putting @LordDarkula's comments inline. The bracket on the next line should just be put right after the bracket on this line so that it reads `= {}`.